R/build-articles.R

```diff
@@ -20,6 +20,52 @@
 #' "Get started". Rmarkdown files in `vignettes/tutorials/` are ignored,
 #' because these are assumed to contain tutorials, see `build_tutorials()`.
 #'
+#' @section Index and navbar:
+#' You can control the index page and navbar with `articles`, which defines a
+#' list of sections. Each section has four fields:
+#'
+#' * `title` (required): title of section, which appears as a heading on the
+#'   articles index.
+#'
+#' * `desc` (optional): An optional markdown description displayed underneath
+#'   the section title.
+#'
+#' * `navbar` (optional): A couple of words used to label this section in
+#'   the navbar. If omitted, this section of vignettes will not appear in the
+#'   navbar. The navbar will automatically include a link to the articles
+#'   index when not all vignettes are listed individually.
+#'
+#' * `contents` (required): a list of article names to include in the
+#'   section. This can either be names of individual vignettes or a
+#'   call to `starts_with()`. The name of a vignette includes is it's
+#'   path under `vignettes` without extension so that the name of the vignette
+#'   found at `vignettes/pizza/slice.Rmd` is `pizza/slice`.
+#'
+#' The title and description of individual vignettes displayed on the index
+#' comes from `title` and `description` fields of the YAML header in the Rmds.
+#'
+#' For example, this yaml might be used for some version of dplyr:
+#'
+#' ```
+#' articles:
+#' - title: Main verbs
+#'   navbar: ~
```

Comment:
One thing that's not clear to me from the PR: where is `~` handled?

Reply:
It's the check for `has_name()` not `is.null()`

Reply:
Ah, got it -- I didn't realize `~` represented a YAML `null`.